### PR TITLE
feat(headers): support known extensions

### PR DESCRIPTION
**Pull request by sanmai-NL**
And only known extensions. Clean up parsing of extensions in general.

### CODE REVIEW

**Comment by sanmai-NL** [comment]
@seanmonstar: could you let me know when you plan to review this?

**Comment by LegNeato** [comment]
This would be a breaking change, right? If so can you add `BREAKING_CHANGE: <description>` to the commit message?

**Comment by seanmonstar** [comment]
Sorry, yes, this would be a breaking change, so it couldn't be part of 0.11. I'll put it in the 0.12 milestone, but if someone needs it immediately, one could just copy the code into a local module and use it instead of importing from hyper.

**Comment by sanmai-NL** [comment]
@LegNeato, @seanmonstar: done. @seanmonstar: okay great, but when is a review planned, if any?

File: src/header/common/cache_control.rs [hunk 1]
```diff
@@ -108,8 +129,8 @@ pub enum CacheDirective {
     /// "s-maxage=delta"
     SMaxAge(u32),
 
-    /// Extension directives. Optionally include an argument.
-    Extension(String, Option<String>)
+    /// Extension directives.
+    Extension(CacheDirectiveExtension),
```
Comment (seanmonstar) [inline src/header/common/cache_control.rs]: The original goal of the `Extension` variant was to allow for new arguments even if the version of hyper doesn't support them (hence being just `String`s). Would it make sense for these new variants to just be on `CacheDirective` itself?

**Comment by seanmonstar** [comment]
As noted in #189, headers are actually going to moved out of hyper proper. So as to not keep this dangling, I'm going to close this. Thanks for the submission, and of course, because the header system works for anything that implements `Header`, anyone can use these changes themselves by creating a new type.

**Comment by sanmai-NL** [comment]
@seanmonstar: would it be okay to merge this PR, for the time being? Nobody is going to look up a stale PR branch to reuse any of this work, so it’d be a waste. Even if you plan to release the redesigned `hyper` version soon, then at least I could pin `hyper` to an older version still has the `header` module, along with this enhancement.

**Comment by sfackler** [comment]
Is there any particular reason you need to pull the header from the Hyper crate specifically?

**Comment by sanmai-NL** [comment]
No there isn’t. I was under the impression that the new headers crate will not be API-compatible with the old one. Apart from that, I assumed that the starting point in terms of headers to reimplement would be the original Hyper headers module, and so that it would matter for everyone whether this PR gets merged or is left depending on further initiative.

**Comment by seanmonstar** [comment]
Unfortunately, it can't just be merged, as it's a breaking change.

**Pull request closed without merging**